appinventor/components/src/com/google/appinventor/components/runtime/Chart.java

```diff
@@ -0,0 +1,457 @@
+     */
+    private ChartView createChartViewFromType(int type) {
+        switch(type) {
+            case ComponentConstants.CHART_TYPE_LINE:
+                return new LineChartView(container.$context());
+            case ComponentConstants.CHART_TYPE_SCATTER:
+                return new ScatterChartView(container.$context());
+            case ComponentConstants.CHART_TYPE_AREA:
+                return new AreaChartView(container.$context());
+            case ComponentConstants.CHART_TYPE_BAR:
+                return new BarChartView(container.$context());
+            case ComponentConstants.CHART_TYPE_PIE:
+                return new PieChartView(container.$context());
+            default:
+                // Invalid argument
+                throw new IllegalArgumentException("type:" + type);
+        }
+    }
+
+    /**
+     * Reinitializes the Chart view by reattaching all the Data
+     * components and setting back all the properties.
+     */
+    private void reinitializeChart() {
+        // Iterate through all attached Data Components and reinitialize them.
+        // This is needed since the Type property is registered only after all
+        // the Data components are attached to the Chart.
+        // This has no effect when the Type property is default (0), since
+        // the Data components are not attached yet, making the List empty.
+        for (ChartDataBase dataComponent : dataComponents) {
+            dataComponent.initChartData();
+        }
+
+        Description(description);
+        BackgroundColor(backgroundColor);
+        LegendEnabled(legendEnabled);
+        GridEnabled(gridEnabled);
+        Labels(labels);
+    }
+
+    /**
+     * Returns the description label text of the Chart.
+     *
+     * @return  description label
+     */
+    @SimpleProperty(
+            category = PropertyCategory.APPEARANCE)
+    public String Description() {
+        return description;
+    }
+
+    /**
+     * Specifies the text displayed by the description label.
+     *
+     * @param text  description
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
+            defaultValue = "")
+    @SimpleProperty
+    public void Description(String text) {
+        this.description = text;
+        chartView.setDescription(description);
+    }
+
+    /**
+     * Returns the chart's background color as an alpha-red-green-blue
+     * integer.
+     *
+     * @return  background RGB color with alpha
+     */
+    @SimpleProperty(
+            category = PropertyCategory.APPEARANCE)
+    public int BackgroundColor() {
+        return backgroundColor;
+    }
+
+    /**
+     * Specifies the chart's background color as an alpha-red-green-blue
+     * integer.
+     *
+     * @param argb  background RGB color with alpha
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_COLOR,
+            defaultValue = Component.DEFAULT_VALUE_COLOR_NONE)
```

Comment:
@ewpatton Wouldn't this disallow usage of the default value, though? Right now the parameter is an integer, and changing it to a String will force the user to put values like "#000000" or similar. If we check against the integer, then every time the user puts in a color that is the default color, it will be changed to NONE.